src/sfloader/fluid_sfont.c

```diff
@@ -700,13 +700,40 @@ int fluid_sample_set_pitch(fluid_sample_t *sample, int root_key, int fine_tune)
  */
 int fluid_sample_validate(fluid_sample_t *sample, unsigned int buffer_size)
 {
+#define EXCLUSIVE_FLAGS (FLUID_SAMPLETYPE_MONO | FLUID_SAMPLETYPE_RIGHT | FLUID_SAMPLETYPE_LEFT)
+    static const unsigned int supported_flags = EXCLUSIVE_FLAGS | FLUID_SAMPLETYPE_LINKED | FLUID_SAMPLETYPE_OGG_VORBIS | FLUID_SAMPLETYPE_ROM;
+
     /* ROM samples are unusable for us by definition */
     if(sample->sampletype & FLUID_SAMPLETYPE_ROM)
     {
         FLUID_LOG(FLUID_WARN, "Sample '%s': ROM sample ignored", sample->name);
         return FLUID_FAILED;
     }
 
+    if(sample->sampletype & ~supported_flags)
+    {
+        FLUID_LOG(FLUID_WARN, "Sample '%s' has unknown flags, which might indicate that it uses a compression that it not supported by fluidsynth", sample->name);
+        return FLUID_FAILED;
+    }
+
+    if((sample->sampletype & EXCLUSIVE_FLAGS) & ((sample->sampletype & EXCLUSIVE_FLAGS) - 1))
+    {
+        FLUID_LOG(FLUID_WARN, "Sample '%s' must be either mono or left or right", sample->name);
+        return FLUID_FAILED;
```

Comment:
Is is really necessary to fail here? Maybe just warn the user and carry on?

Reply:
Idk. Such a flag combination is not defined by the spec. But the spec also doesn't seem to specify how to treat samples with those invalid combinations. If we warn only, should we force to sample to MONO or simply leave `sample->sampletype` unchanged?

Reply:
Not sure... I think it depends on if those flag combinations actually cause a real problem. If we rely on consistent flags and something breaks otherwise, we should correct. Otherwise leave as is?

Reply:
Ok. When looking through the code, I cannot find a place where the `sampletype` actually makes a difference. This may become relevant if we ever implement #61. So, let's leave it unchanged for now. For both cases. Will apply the changes.

Reply:
Sounds good!